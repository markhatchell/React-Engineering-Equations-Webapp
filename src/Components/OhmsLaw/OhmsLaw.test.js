import React from 'react';
import renderer from 'react-test-renderer';
import {shallow, configure} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import OhmsLawComponent from "./OhmsLaw";

configure({ adapter: new Adapter() });

const dom = renderer.create(
  <OhmsLawComponent />,
);

describe('OhmsLawComponent', () => {

  test('snapshot matches', () => {
    let tree = dom.toJSON();
    expect(tree).toMatchSnapshot();
  });

  const component = shallow(<OhmsLawComponent />);
  const instance = component.instance();


  test('voltage should be 10', () => {
    component.setState(Object.assign(
      {},
      component.state(),
      {
        voltage: "10"
      }
    ));
    expect(component.state("voltage")).toBe("10");
  });

  test('current should be .5', () => {
    component.setState(Object.assign(
      {},
      component.state(),
      {
        voltage: "10",
        current: ".5"
      }
    ));
    expect(component.state("current").toString()).toBe(".5");
  });

  test('calculated resistance should be 20', () => {
    component.setState(Object.assign(
      {},
      component.state(),
      {
        voltage: "10",
        current: ".5"
      }
    ));
    instance.calculateOtherValues();
    expect(component.state("resistance").toString()).toBe("20");
  });

  test('calculated watts should be 5', () => {
    component.setState(Object.assign(
      {},
      component.state(),
      {
        voltage: "10",
        current: ".5"
      }
    ));
    instance.calculateOtherValues();
    expect(component.state("watts").toString()).toBe("5");
  });

});
