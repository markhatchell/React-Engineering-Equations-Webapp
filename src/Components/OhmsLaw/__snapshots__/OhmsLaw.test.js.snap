// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OhmsLawComponent snapshot matches 1`] = `
<div
  className="equationBox"
>
  <h3>
    Ohms Law
  </h3>
  <div
    className="equationBoxInner"
  >
    <p>
      Supply at least two of the values to calculate the third.
    </p>
    <div
      className="tableContainer"
    >
      <form
        onSubmit={[Function]}
      >
        <table
          className="table"
        >
          <tbody>
            <tr>
              <td
                align="left"
              >
                <label
                  htmlFor="omhslaw-I"
                >
                  I (Current)
                </label>
              </td>
              <td>
                <input
                  autoComplete="off"
                  disabled={false}
                  id="omhslaw-I"
                  onChange={[Function]}
                  tabIndex="1"
                  value=""
                />
              </td>
            </tr>
            <tr>
              <td
                align="left"
              >
                <label
                  htmlFor="omhslaw-V"
                >
                  V (Volts)
                </label>
              </td>
              <td>
                <input
                  autoComplete="off"
                  disabled={false}
                  id="omhslaw-V"
                  onChange={[Function]}
                  tabIndex="2"
                  value=""
                />
              </td>
            </tr>
            <tr>
              <td
                align="left"
              >
                <label
                  htmlFor="omhslaw-R"
                >
                  R (Resistance)
                </label>
              </td>
              <td>
                <input
                  autoComplete="off"
                  disabled={false}
                  id="omhslaw-R"
                  onChange={[Function]}
                  tabIndex="3"
                  value=""
                />
              </td>
            </tr>
            <tr>
              <td
                align="left"
              >
                <label
                  htmlFor="omhslaw-W"
                >
                  P (Watts)
                </label>
              </td>
              <td>
                <input
                  autoComplete="off"
                  disabled={false}
                  id="omhslaw-W"
                  onChange={[Function]}
                  tabIndex="4"
                  value=""
                />
              </td>
            </tr>
            <tr>
              <td
                align="left"
              >
                <button
                  onClick={[Function]}
                  tabIndex="6"
                  type="button"
                >
                  Clear
                </button>
              </td>
              <td
                align="right"
              >
                <button
                  tabIndex="5"
                  type="submit"
                >
                  Calculate
                </button>
              </td>
            </tr>
          </tbody>
        </table>
      </form>
    </div>
  </div>
  <img
    alt="Omh's Law formula Wheel"
    className="Omh's Law Formula Wheel"
    src="/FormulaWheelElectronics.gif"
  />
</div>
`;
